src/linear_samplers.jl

```diff
@@ -0,0 +1,186 @@
+############################################################################################
+## This file is part of RLinearAlgebra.jl
+##
+## Overview: abstractions and methods for sampling, sketching or selecting from linear
+## systems of equations.
+##
+## Contents
+## - Abstract Types
+## - `sample` function documentation
+## - Vector Row Sampler/Sketch/Selector
+## - Vector Column Sampler/Sketch/Selector
+## - Block Row Sampler/Sketch/Selector
+## - Block Column Sampler/Sketch/Selector
+## - Export Statements
+##
+############################################################################################
+
+# Dependencies: Random, Distributions, LinearAlgebra
+
+#############################################
+# Abstract Types
+#############################################
+
+"""
+    LinSysSampler
+
+Abstract supertype for sampling, sketching or deterministically selecting components
+    of a linear system.
+
+# Aliases
+- `LinSysSketch`
+- `LinSysSelect`
+"""
+abstract type LinSysSampler end
+LinSysSketch = LinSysSampler
+LinSysSelect = LinSysSampler
+
```

Comment:
@dmaldona effectively, yes. We can think of a select as a sample generated with probability one. We can think of a sketch as a special case of sample. I gave them aliases because the same question might come up from others even though they are logically all the same.

Reply:
@vp314 I see! Thanks!